src/Apps/Sell/Components/UploadPhotosForm.tsx

```diff
@@ -0,0 +1,129 @@
+import { useToasts } from "@artsy/palette"
+import { useAddAssetToConsignmentSubmission } from "Apps/Consign/Routes/SubmissionFlow/Mutations"
+import { PhotosFormValues } from "Apps/Sell/Routes/PhotosRoute"
+import { PhotoDropzone } from "Components/PhotoUpload/Components/PhotoDropzone"
+import {
+  Photo,
+  getErrorMessage,
+  normalizePhoto,
+  uploadSubmissionPhoto,
+} from "Components/PhotoUpload/Utils/fileUtils"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { getENV } from "Utils/getENV"
+import createLogger from "Utils/logger"
+import { useFormikContext } from "formik"
+import { useCallback, useEffect } from "react"
+import { FileRejection } from "react-dropzone"
+
+const logger = createLogger("Sell/UploadPhotosForm.tsx")
+
+export const UploadPhotosForm: React.FC = () => {
+  const { isLoggedIn, relayEnvironment } = useSystemContext()
+  const { submitMutation: addAsset } = useAddAssetToConsignmentSubmission()
+  const { setFieldValue, values } = useFormikContext<PhotosFormValues>()
+  const { sendToast } = useToasts()
+
+  const handlePhotoUploadingProgress = useCallback(
+    (photo: Photo) => progress => {
+      photo.progress = progress
+      setFieldValue("photos", values.photos)
+    },
+    [values.photos, setFieldValue]
+  )
+
+  const uploadImage = useCallback(
```

Comment:
What's the reason we are using `useCallback` here?

Reply:
I guess I can move these functions to `useEffect` below and add their dependencies to the dependency array. I preferred to use `useCallback` instead (I guess it should be the same, no?) @olerichter00